### PR TITLE
Add getters and setters for `InputAxis` and `ButtonSettings`

### LINKED ISSUE #3418: Validate ButtonSettings and AxisSettings using getter / setter pattern
## Problem

AxisSettings and GamepadButtonSettings can easily end up in a nonsensical or broken state, as the values are fully pub.

## Proposed solution

Correctness should be enforced through `get` and `new` methods, and the fields should not be public.

Correctness rules:
1. The ordering must be correct; check the default settings for guidance.
2. All values must be in the appropriate range.

### PR DESCRIPTION
**Pull request by mfdorst**
Fixes #3418

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Thanks for contributing, and the great questions!

I've replied to your questions over in #3446, could you apply them to this PR if you agree with them? :)

**Comment by mfdorst** [comment]
@alice-i-cecile I do agree with them, thanks. I'll make sure to run clippy this time too lol. Not sure why `CI / build (stable, ubuntu-latest) (pull_request)` failed though. From the error it looks like a connectivity problem maybe?

**Comment by mfdorst** [comment]
Fixed up `AxisSettings`. Will get to `ButtonSettings` soon.

**Comment by alice-i-cecile** [comment]
Could you add `ButtonSettings` to this PR's title too so then it's easier to find when skimming?

**Review by alice-i-cecile — APPROVED** [review]
Awesome, this looks fantastic. What a nice improvement!

File: crates/bevy_input/src/gamepad.rs [hunk 1]
```diff
@@ -129,72 +141,380 @@ impl GamepadSettings {
...
 
 impl AxisSettings {
-    fn filter(&self, new_value: f32, old_value: Option<f32>) -> Option<f32> {
-        let new_value = if new_value <= self.positive_low && new_value >= self.negative_low {
-            0.0
-        } else if new_value >= self.positive_high {
-            1.0
-        } else if new_value <= self.negative_high {
-            -1.0
+    /// Creates a new AxisSettings instance
```
Comment (alice-i-cecile) [inline crates/bevy_input/src/gamepad.rs:303]:
```suggestion
    /// Creates a new [`AxisSettings`] instance
```

File: crates/bevy_input/src/gamepad.rs [hunk 2]
```diff
@@ -2,6 +2,18 @@ use crate::{Axis, Input};
 use bevy_app::{EventReader, EventWriter};
 use bevy_ecs::system::{Res, ResMut};
 use bevy_utils::{tracing::info, HashMap, HashSet};
+use thiserror::Error;
+
+/// Errors that occur when setting settings for gamepad input
```
Comment (IceSentry) [inline crates/bevy_input/src/gamepad.rs:7]: maybe "connfiguring settings" or "setting up gamepad input". "setting settings" feels weird

Comment (Weibye) [inline crates/bevy_input/src/gamepad.rs:7]:
```suggestion
/// Errors that occur when attempting to incorrectly configure gamepad input
```

**Review by Weibye — COMMENTED** [review]
Some minor nits to doc-comments, otherwise this looks really solid :)

**Comment by alice-i-cecile** [comment]
@mfdorst do you have time to get back to this? This is almost ready to merge :)

**Comment by alice-i-cecile** [comment]
Closing in favor of the rebased #6050.

**Pull request closed without merging**